src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -211,6 +211,59 @@ protected override ModuleFieldLayout CreateValueFromKey(EcmaModule module)
                     typeOffsets: typeOffsets);
             }
 
+            private void GetElementTypeInfoGeneric(
+                EcmaModule module,
+                FieldDesc fieldDesc,
+                EntityHandle valueTypeHandle,
+                bool moduleLayout,
+                ref int alignment,
+                ref int size,
+                ref bool isGcPointerField,
+                ref bool isGcBoxedField)
+            {
+                TypeDesc fieldType = fieldDesc.FieldType;
+
+                if (fieldType.IsPrimitive)
+                {
+                    size = fieldType.GetElementSize().AsInt;
+                    alignment = size;
+                }
+                else if (fieldType.IsValueType)
+                {
+                    if (IsTypeByRefLike(valueTypeHandle, module.MetadataReader))
```

Comment:
If you move the `else if (fieldType.IsByRef || fieldType.IsByRefLike || fieldType.IsByReferenceOfT)` check before `else if (fieldType.IsValueType)`, this will become unreachable and can be deleted (now the other IsByRefLike check is unreachable instead).

Reply:
The IsTypeByRefLike check can be deleted because it's unreachable (fieldType.IsByRefLike above covers it).